### ISSUE TITLE
Training adapter for mutation effect prediction on SaprotHub

### ISSUE THREAD

**Issue by anton-bushuiev** [westlake-repl/SaprotHub#78]
Hi @LTEnjoy!

Thank you for your amazing work with SaProt and SaProt Hub. Can you please help me to figure out if it is possible to use the [SaProt Hub Colab notebook](https://colab.research.google.com/github/westlake-repl/SaprotHub/blob/main/colab/SaprotHub.ipynb#) for the following use case: I have a protein and ~200 labeled mutations. I want to fine-tune SaProt on this small dataset and use it predict effects of other mutations. However, in the notebook I cannot see a suitable task:

<img width="452" alt="image" src="https://github.com/user-attachments/assets/3270c664-1c49-4664-b520-4123f5304220">

Thank you!
Anton

**Comment by LTEnjoy** [comment]
Hi Anto!

Thank you for your interest in our work! In your case, we would recommend use ``Protein-level Regression``, which means you input a protein sequence and the model will predict a single value about its property (in your case it's mutational effects).

Then you could choose to train the model on your dataset.
![image](https://github.com/user-attachments/assets/6571882b-d1f8-49ea-90a6-95082bcf26f7)

You will be asked to uplaod a csv file containing your data. It looks like this:
![1729562291188](https://github.com/user-attachments/assets/9ece7394-a0df-4680-9d98-2bd4ab356afd)

For more details, please refer to https://github.com/westlake-repl/SaprotHub/wiki/2.1:-Train-your-model.

BTW, currently we are building a more concise version of SaprotHub. We will re-organize its modules and make it more understandable, which I think should be more helpful for you :)

Best,
Jin

**Comment by anton-bushuiev** [comment]
Hi Jin,

Thank you very much for the quick response! I will experiment with protein-level regression.

I am looking forward to the new version of SaProt Hub :)!

Best,
Anton


P.S. Just an idea that could be relevant: for mutations, it could also be interesting to fine-tune SaProt directly via log-odds ratio (as you do in zero-shot setup for experiments with ProteinGym) as opposed to a general protein-level regression head. This was done in [AlphaMissense](https://www.science.org/doi/10.1126/science.adg7492) and, for example, in our [PPIformer](https://arxiv.org/abs/2310.18515), where fine-tuning through log odds worked much better than general protein-level regression (Table 6 in the paper).

**Comment by LTEnjoy** [comment]
Hi Anton,

Thanks for the nice suggestion! Perhaps this could be done when users provide their training data, i.e. setting the log-odds ratio as supervised labels rather than fitness scores. We can also consider displaying more prompts to guide users for how to more effectively fine-tune their models.

Best regards,
Jin

**Comment by anton-bushuiev** [comment]
Sounds great! Thanks again for the nice work!

Best,
Anton
